diff -ru a/Alc/bformatdec.h b/Alc/bformatdec.h
--- a/Alc/bformatdec.h	2018-10-11 23:05:31.000000000 +0100
+++ b/Alc/bformatdec.h	2026-09-27 10:35:04.919872596 +0100
@@ -24,9 +24,9 @@
 /* NOTE: These are scale factors as applied to Ambisonics content. Decoder
  * coefficients should be divided by these values to get proper N3D scalings.
  */
-const ALfloat N3D2N3DScale[MAX_AMBI_COEFFS];
-const ALfloat SN3D2N3DScale[MAX_AMBI_COEFFS];
-const ALfloat FuMa2N3DScale[MAX_AMBI_COEFFS];
+extern const ALfloat N3D2N3DScale[MAX_AMBI_COEFFS];
+extern const ALfloat SN3D2N3DScale[MAX_AMBI_COEFFS];
+extern const ALfloat FuMa2N3DScale[MAX_AMBI_COEFFS];
 
 
 struct AmbDecConf;
diff -ru a/Alc/helpers.c b/Alc/helpers.c
--- a/Alc/helpers.c	2018-10-11 23:05:31.000000000 +0100
+++ b/Alc/helpers.c	2026-09-27 10:34:34.337028763 +0100
@@ -1010,7 +1010,8 @@
 #ifdef _WIN32
     if(RTPrioLevel > 0)
         failed = !SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
-#elif defined(HAVE_PTHREAD_SETSCHEDPARAM) && !defined(__OpenBSD__)
+#elif defined(HAVE_PTHREAD_SETSCHEDPARAM) && !defined(__OpenBSD__) && !defined(__riscos__)
+    /* (RISC OS: UnixLib has no real-time scheduling or sched_get_priority_min) */
     if(RTPrioLevel > 0)
     {
         struct sched_param param;
